src/ast/transform/ADTtoRecords.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file ADTtoRecords.h
+ *
+ * Defines the desugaring of ADTs to records.
+ *
+ * Each record has a form [branch_id, [arguments]].
+ * Branch ID is given by a lexicographical ordering of branches within an ADT.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "ast/transform/Transformer.h"
+
+namespace souffle {
+
+class ADTtoRecords : public AstTransformer {
```

Comment:
The classname should either end in `Transformer` or `Checker`. We're trying to get these names to be more consistent.